packages/nextjs/src/config/index.ts

```diff
@@ -12,17 +12,18 @@ export function withSentryConfig(
   userNextConfig: ExportedNextConfig = {},
   userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
 ): NextConfigFunction | NextConfigObject {
-  const newWebpackConfig = constructWebpackConfigFunction(userNextConfig, userSentryWebpackPluginOptions);
-
   // If the user has passed us a function, we need to return a function, so that we have access to `phase` and
   // `defaults` in order to pass them along to the user's function
   if (typeof userNextConfig === 'function') {
-    return (phase: string, defaults: { defaultConfig: { [key: string]: unknown } }): NextConfigObject => ({
-      ...userNextConfig(phase, defaults),
-      webpack: newWebpackConfig,
-    });
+    return function(phase: string, defaults: { defaultConfig: { [key: string]: unknown } }): NextConfigObject {
```

Comment:
Nit: prefer an arrow function if we are just returning an anonymous func

Reply:
I had one, then this `return` was getting visually muddied with the `return` inside the returned function, so I changed it to this, to make it clearer/easier to differentiate.